services/webnn/ort/graph_builder_ort.cc

```diff
@@ -241,14 +272,15 @@ std::string GraphBuilderOrt::GetOperandName(uint64_t operand_id) {
   }
 }
 
-template <typename T>
-std::string GraphBuilderOrt::CreateInitializer(base::span<const uint32_t> shape,
-                                               base::span<const T> data,
-                                               OperandDataType data_type) {
+template <typename DataType>
+  requires internal::IsSupportedTensorType<DataType>
+std::string GraphBuilderOrt::CreateInitializer(
+    base::span<const uint32_t> shape,
+    base::span<const DataType> data) {
```

Comment:
Could you please explain why it is safe to call `base::as_byte_span(data)` when `data` is `base::span<Float16>`?